litr-filters/src/main/java/com/linkedin/android/litr/filter/audio/VolumeFilter.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 LinkedIn Corporation
+ * All Rights Reserved.
+ *
+ * Licensed under the BSD 2-Clause License (the "License").  See License in the project root for
+ * license information.
+ */
+package com.linkedin.android.litr.filter.audio
+
+import android.media.MediaFormat
+import androidx.annotation.FloatRange
+import com.linkedin.android.litr.codec.Frame
+import com.linkedin.android.litr.filter.BufferFilter
+import kotlin.math.pow
+
+private const val SAMPLE_SIZE = 2 // audio sample size on Android is 2 bytes per channel
+private const val BASE = 10.0 // volume is logarithmic, we will use base 10
+
+/**
+ * An audio filter that changes the audio track volume:
+ * - 1 will keep the volume "as is"
+ * - 0 will silence the track
+ * - value <1 will lower the volume
+ * - value >1 will increase it. One has to be careful with these, since large values may result in distortion.
+ */
+class VolumeFilter(@FloatRange(from = 0.0) private val volume: Double) : BufferFilter{
```

Comment:
`Filter` doesn't seem associated with audio processing (thinking about sound effect, boost, mixer, etc). Possible better naming?

Reply:
I couldn't really come up with better common naming for audio and video frame modifications. I guess "filter" works better with video because it became a common term with video/photo filters in popular apps. With audio "filters" have more of a "signal processing" meaning low/high pass filter, etc. Maybe we can keep "filter" terminology within LiTr?